src/renderers/dom/shared/DOMPropertyOperations.js

```diff
@@ -188,6 +188,10 @@ var DOMPropertyOperations = {
     }
     if (value == null) {
       node.removeAttribute(name);
+    } else if (value === true) {
+      node.setAttribute(name, '');
+    } else if (value === false) {
+      node.removeAttribute(name);
```

Comment:
I don't know if there is any cross over with #9806, but I'm curious, if we move forward with this, how `value` should be treated. 

We have [a special mutation method for `value`](https://github.com/facebook/react/blob/master/src/renderers/dom/shared/HTMLDOMPropertyConfig.js) that might need to be synced up with this new logic, but as far as I see it, `value={true}` should stringify to `"true"`, which is what would happen if you mutated the input directly.